### DISCUSSION TITLE
Querying 1-to-n relation in graphql on Web side

### DISCUSSION THREAD

**Discussion by maverpolo022** [prisma/prisma#16146] [Q&A]
I have this model
```
model Author {
    id String @id @default(cuid())
    Posts Post[]
}

model Post {
    ...
   author Author @relation(fields: [authorId], references: [id])
   authorid String
}
```
On api side, i am able to fetch all posts of a user by
```
const q = await db.Author.findUnique({
      where: { id: 12345 },
        include: {
          Posts: true,
        }
    })
    console.log(q)
```
but on the frontend side, I am not able to do so using graphql

I tried
```
export type FindById = { 
  __typename?: 'Query', 
  author?: {
     __typename?: 'author', 
     id: string, 
    } | null 
  };

gql`
query FindById($id: String!) {
    author: author(id: $id) {
      id
      Posts
    }
}
```
I don't know what should i include in type or query to make this work

**Comment by nurul3101** [comment]
Hello @mr-loop-1 👋 
If I understand correctly your resolvers are working as expected on the backend side. Are you receiving any specific errors on the frontend when trying to query?

**Reply by maverpolo022** [reply]
Hi @nurul3101 It says that field `Posts` does not exist in Author. I am very new to Graphql and maybe the `type` or query need some modification. The problem seems to be from the fact the `Posts` does not really exist in the table as a field so maybe I cannot query it directly like other fields.

**Reply by nurul3101** [reply]
Hey @mr-loop-1, Apologies for the delay in getting back to you. Is your code open source? If yes, can you share the repository link so that I can reproduce locally and update it to make it working? Also, did you had a chance to look at our GraphQL [examples](https://github.com/prisma/prisma-examples/tree/latest/typescript), they should help you in getting started.

**Reply by vladi-stevanovic** [reply]
👋 Hi @mr-loop-1 , did you perhaps have a chance to check Nurul's recommendation? Let us know if you still have any questions! 

Or, if you've found a solution, we'd love to know how you solved this problem. Feel free to post it in this thread and "Mark as answer" if this discussion is now solved!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
